src/plugins/footnotes/index.ts

```diff
@@ -244,31 +276,56 @@ export default (props: PluginProps) => {
       },
 
       apply(tr, value, oldState, newState): PluginState {
+        const prevState = footnotesKey.getState(oldState)
+        if (!tr.docChanged && prevState) {
+          return prevState
+        }
         return buildPluginState(newState.doc)
       },
     },
 
     appendTransaction(transactions, oldState, newState) {
-      const { nodes: oldInlineFootnoteNodes } = footnotesKey.getState(
+      const { inlineFootnotes: oldInlineFootnoteNodes } = footnotesKey.getState(
         oldState
       ) as PluginState
 
-      const { nodes: inlineFootnoteNodes, labels } = footnotesKey.getState(
-        newState
-      ) as PluginState
-
-      if (isEqual(inlineFootnoteNodes, oldInlineFootnoteNodes)) {
+      const {
+        inlineFootnotes: inlineFootnoteNodes,
+        footnotes,
+        labels,
+        footnoteElement,
+      } = footnotesKey.getState(newState) as PluginState
+
+      const prevIds = oldInlineFootnoteNodes.map(([node]) => node.attrs.rids)
+      const newIds = inlineFootnoteNodes.map(([node]) => node.attrs.rids)
+      const initTransaction = transactions.find((t) => t.getMeta('INIT'))
+
+      if (
+        !footnoteElement ||
+        (!initTransaction && (isEqual(prevIds, newIds) || newIds.length < 2))
+      ) {
         return null
       }
 
       const { tr } = newState
 
+      const footnotesReordered: FootnoteNode[] = []
+      const footnotesRest = new Map(footnotes)
+
       inlineFootnoteNodes.forEach(([node, pos]) => {
         const footnote = node as InlineFootnoteNode
         const contents = footnote.attrs.rids
-          .map((rid) => labels.get(rid))
+          .map((rid) => {
+            const currentfNode = footnotesRest.get(rid)
+            if (currentfNode) {
+              footnotesReordered.push(currentfNode[0])
+              footnotesRest.delete(rid)
+            }
+            return labels.get(rid)
+          })
           .join('')
 
+        // displaying indices
```

Comment:
Is it possible to avoid updating the inline_footnote contents? what's displayed can be derived from the plugin state in a view.

Reply:
Not sure what you mean. Are you suggesting to use decorations instead of updating a node attribute?

Reply:
yeah, that's what we do for citations right?

Reply:
Aren't we relying on the contents during jats export? See: https://github.com/Atypon-OpenSource/manuscripts-transform/blob/master/src/jats/jats-exporter.ts#L1073

Reply:
That should also be changed (not in this ticket).

Reply:
I should add that we don't need to make this change now, but I do think we should do that eventually.

Reply:
I see. Indeed that should be done in a different ticket if needs to be.